File: frontend/src/microservices/tour-group/objects/TourGroupBlacklistResponse.ts
// 本文件定义 tour-group 模块的 `TourGroupBlacklistResponse`，作为列表响应数据并提供 JSON 编解码。

export type TourGroupBlacklistResponse = {
  blacklistId: string
  groupId: string
  userId: string
  blacklistedByUserId: string
  reason: string
  createdAt: string
}

export const tourGroupBlacklistResponseFromJson = (json: string): TourGroupBlacklistResponse =>
  JSON.parse(json) as TourGroupBlacklistResponse

export const tourGroupBlacklistResponseToJson = (value: TourGroupBlacklistResponse): string =>
  JSON.stringify(value)

File: backend/src/main/scala/microservices/tour-group/objects/TourGroupBlacklistResponse.scala
// TourGroupPlannerModels 定义团体游模块的请求和响应模型。

package com.typesafe.travel.tourgroup.domain

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class TourGroupBlacklistResponse(
    blacklistId: String,
    groupId: String,
    userId: String,
    blacklistedByUserId: String,
    reason: String,
    createdAt: String
)
object TourGroupBlacklistResponse:
  given sourceEncoder: Encoder[TourGroupBlacklistResponse] = deriveEncoder
  given sourceDecoder: Decoder[TourGroupBlacklistResponse] = deriveDecoder
